### PR TITLE
Add button designdoc content

### PR DESCRIPTION
**Pull request by stacymcauliffe**
Added new design docs content for button component including usage and content guidelines

### CODE REVIEW

**Comment by stacymcauliffe** [comment]
@mcarrano @LHinson - here is my first PR. It won't let me add reviewers, maybe I need to be added to this repo? Hopefully I followed all the steps correctly. Let me know! :)

**Comment by LHinson** [comment]
Congrats on your first PR!!! 

bitmoji

I wasn't quite sure if I was supposed to review this as final or not but went ahead with some nitpicky comments :)

File: src/patternfly/components/Button/docs/design.md [hunk 1]
```diff
@@ -1,4 +1,59 @@
...
+#### Primary
+Use primary buttons for the main action you want a user to take on a page. In general, try to limit primary buttons to one per page.
+
+#### Secondary
+Use secondary buttons for secondary actions a user can take. For example, a cancel button would be secondary to a submit button on a form or modal.
+
+#### Tertiary
+Use tertiary buttons for any action a user can take on a page that is not primary or secondary. For example, actions a user might take to edit items in a list could be represented as tertiary buttons.
+
+#### Danger
```
Comment (mcarrano) [inline src/patternfly/components/Button/docs/design.md]: In the image these are labeled as "Destructive" buttons, which is consistent with how we've named these before. Are you recommending we change that @stacymcauliffe ?

File: src/patternfly/components/Button/docs/design.md [hunk 2]
```diff
@@ -1,4 +1,59 @@
...
+For actions a user can take that are potentially destructive or difficult/impossible to undo like deleting or removing user data
+
+### Button States
```
Comment (mcarrano) [inline src/patternfly/components/Button/docs/design.md]: I'm wondering if this section is really needed. These states should be baked into the code and for the most part designers don't need to be concerned with this. Use of disabled buttons might be the exception. However when and where to use a disabled button is so contextual that I'm not sure there's much to say here.

File: src/patternfly/components/Button/docs/design.md [hunk 3]
```diff
@@ -1,4 +1,59 @@
...
+![Buttons on modals](img/button_on_form.png)
+
+#### Buttons on modals
+* Align submit buttons to the bottom right of the modal.
+* Place the primary button to the right of the cancel button.
+* Align submit buttons with right-edge of input fields.
+
+![Buttons on modals](img/button_on_modal.png)
+
+## Content Guidelines
```
Comment (mcarrano) [inline src/patternfly/components/Button/docs/design.md]: Having this section is great. In my past experience, I was always told to allow labels to grow about 30% to account for localization. This is not something that we've really dealt with in PatternFly and in fact does not seem to be a huge issue for Red Hat in general. Not sure if we should say anything about this here. What do you think @LHinson ?

**Pull request merged**